app/services/course/assessment/submission/statistics_download_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+require 'csv'
+class Course::Assessment::Submission::StatisticsDownloadService
+  include ApplicationFormattersHelper
+
+  class << self
+    # Downloads the statistics and zip them.
+    #
+    # @param [User] current_user The current user downloading the statistics.
+    # @param [Course::Assessment] assessment The assessments to download  from.
+    # @param [Array<Integer>] submission_ids The ids of the submissions to download statistics for
+    # @return [String] The path to the zip file.
+    def download(current_user, assessment, submission_ids)
+      service = new(current_user, assessment, submission_ids)
+      ActsAsTenant.without_tenant do
+        service.send(:download_to_base_dir)
+      end
+    end
+  end
+
+  private
+
+  def initialize(current_user, assessment, submission_ids)
+    @current_user = current_user
+    @assessment = assessment
+    @submission_ids = submission_ids
+    @base_dir = Dir.mktmpdir('coursemology-statistics-')
+  end
+
+  def download_to_base_dir
```

Comment:
😄 pehaps we can rename this method to `generate_csv_report`? (or some other names which reveal the intent)

Reply:
OK, will rename the method.